pkg/cloud/metadata_k8s.go

```diff
@@ -32,6 +33,21 @@ var DefaultKubernetesAPIClient = func() (kubernetes.Interface, error) {
 	return clientset, nil
 }
 
+// Get testing kubernetes APIclient
+var TestingKubernetesAPIClient = func() (kubernetes.Interface, error) {
```

Comment:
why this function defined here?

Reply:
Since the test cases are running in the cluster(not inside the pod), Need to get config from the config file **/root/.kube/config** as below
```
config, err := clientcmd.BuildConfigFromFlags("", "/root/.kube/config")
```

DefaultKubernetesAPIClient loads the config from rest API as below (as running inside csi driver pod):
```
config, err := rest.InClusterConfig()
```

Hence, 2 separate clients needed based on where our code runs.
Default client is used in csi driver, Testing client is used in e2e testing.

Reply:
Conversation outdated as the approach changed.
CloudInstanceId is retrieved from the node labels using test suites library instead of using metadata library.